@ecomplus/storefront-components/src/js/TheProduct.js

```diff
@@ -232,18 +233,31 @@ export default {
     discount () {
       const { body } = this
       const priceValue = this.fixedPrice || getPrice(body)
-      return checkOnPromotion(body)
+      return checkOnPromotion(body) || (body.price > priceValue)
         ? Math.round(((body.base_price - priceValue) * 100) / body.base_price)
         : 0
     },
 
-    isOnSale () {
+    mockNewPromoDate () {
       const { body } = this
+      const newPromoDate = { ...body }
+      if (this.alwaysShowTimer) {
+        const tomorrow = new Date(new Date().getTime() + 86400000).setHours(0, 0, 0, 0)
+        newPromoDate.price_effective_date = {}
+        newPromoDate.price_effective_date.end = new Date(tomorrow).toISOString()
+        return newPromoDate
+      } else {
+        return body
+      }
+    },
```

Comment:
Não faz sentido criar esse computed só pra isso

Reply:
Piorou clonar o objeto do produto todo 😬

Reply:
Você só precisava deixar o `isOnSale` true caso `hasPromotionTimer === 'always'`, essa condição pode ser removida: https://github.com/ecomplus/storefront/blob/41108ecb629c8f8ecd0f4f5e15db2acbcb1733fd/%40ecomplus/storefront-components/src/js/TheProduct.js#L575 (se o produto realmente estiver em promoção isso é verdadeiro, caso contrário se chegou aí é porque o `isOnSale` for forçado com a prop lá...)